azure/scope/machinepoolmachine.go

```diff
@@ -274,14 +287,23 @@ func (s *MachinePoolMachineScope) UpdateStatus(ctx context.Context) error {
 		node    *corev1.Node
 		err     error
 	)
-	if nodeRef == nil || nodeRef.Name == "" {
+
+	switch {
+	case s.ProviderID() == "":
+		conditions.MarkFalse(s.AzureMachinePoolMachine, clusterv1.MachineNodeHealthyCondition, clusterv1.WaitingForNodeRefReason, clusterv1.ConditionSeverityInfo, "")
+	case nodeRef == nil || nodeRef.Name == "":
 		node, err = s.workloadNodeGetter.GetNodeByProviderID(ctx, s.ProviderID())
-	} else {
+		if err != nil && !apierrors.IsNotFound(err) {
+			// TODO: what should we set the condition to when node == nil and err == nil?
```

Comment:
@CecileRobertMichon @devigned Right now we call `conditions.MarkFalse(s.AzureMachinePoolMachine, clusterv1.MachineNodeHealthyCondition, clusterv1.NodeProvisioningReason, clusterv1.ConditionSeverityInfo, "")` when `err != nil && !apierrors.IsNotFound(err)`, but we don't set anything when `node == nil` and `err == nil`. Which condition should we mark in the latter case? I'm not sure what `node == nil` and `err == nil` is meant to represent.

Reply:
If `node == nil`, then the node associated to the machine pool machine was not found, likely it has not joined the cluster yet. 

`err == nil` signifies there was no error when communicating with the API server.

The gist of what is going on here is that we are attempting to determine the state of the infrastructure and the health / presence of the corresponding node in the cluster.

Reply:
Okay thanks for the info. So in this case I think if `node == nil` then we can set it to `clusterv1.NodeProvisioningReason`. In the latter case where `err != nil`, would we want to set a condition reason here? We don't want the condition to be ready, but I'm not sure what we can say about the node based on that info.

Reply:
I don't know what reason to put there either. If we have a failure interacting with the API server, we should exit with error and require the reconcile (default behavior).

The reason could be unknown, perhaps.

Reply:
As an aside, I never liked this func. There's too much going on, too many conditionals. I didn't put the time into refactoring. However, if you have it all queued up in your head, it might be a good opportunity to restructure a bit to make it easier on the next person.

Reply:
Yeah I can look into it and see if we can refactor a bit. In general, if we have a failure with the API server, would we want to mark the condition at all?